### DISCUSSION TITLE
Capacitor with legacy ES5 (AngularJS) app

### DISCUSSION THREAD

**Discussion by aderowbotham** [ionic-team/capacitor#3493] [Q&A]
I am a bit stuck following the Capacitor guide for enabling core plugins.

My existing app is written in AngularJS (v1.7.9, LTS) and I am using ES5 syntax. I am not using a module loader so **cannot** do this (e.g. Local Notifications)

    import { Plugins } from '@capacitor/core';
    const { LocalNotifications } = Plugins;

I have my app running in XCode.

I set `"bundledWebRuntime": true,` in **capacitor.config.json** before calling **npx cap copy**. I can see that this has copied **capacitor.js** to my web content directory, and I am loading it with a `<script>` tag, similar to Cordova. 

But –  seeing as there is no config.xml – how do I now load a specific plugin such as **LocalNotifications**?

There does not appear to be any documentation for working with Capacitor core Plugins this way. Are they ES5 compatible?

**Comment by aderowbotham** [comment]
Further to this, it appears the plugins are exposed in a variable called **capacitorExports**. This doesn't seem to be documented.

Running in the iOS Simulator I can view the LocalNotifications plugin like this:

    if(capacitorExports){
          var LocalNotifications = capacitorExports.LocalNotifications
          console.debug("LocalNotifications = ", LocalNotifications);
    }

This gives the output:

    LocalNotifications =  
      t
        config: {name: "LocalNotifications", platforms: ["web"]}
        listeners: {}
        loaded: false
        pending: [Object, Object] (2)
        windowListeners: {}


I note that the platforms only contains **web** and loaded is **false**. Why is it not loaded?

**LocalNotifications.schedule()** seems to fail silently (no notification appears)

**LocalNotifications.areEnabled()** throws an error:

    LocalNotifications.areEnabled().then(
      function resolved(resp){
            $log.debug("resolved: ", resp);
      }
    );

outputs:

```
[Error] Can't find variable: Notification
capacitor://localhost/capacitor.js:2
	capacitorConsole (user-script:2:78)
	(anonymous function) (vendor.min.kek06jwq.js:107)
	(anonymous function) (vendor.min.kek06jwq.js:177:348)
	(anonymous function) (vendor.min.kek06jwq.js:174)
	(anonymous function) (vendor.min.kek06jwq.js:60)
[Error] Can't find variable: Notification
capacitor://localhost/capacitor.js:2 – "Possibly unhandled rejection: {}"
	capacitorConsole (user-script:2:78)
	(anonymous function) (vendor.min.kek06jwq.js:107)
	g (vendor.min.kek06jwq.js:147:226)
	$digest (vendor.min.kek06jwq.js:159:187)
	$apply (vendor.min.kek06jwq.js:163)
	(anonymous function) (vendor.min.kek06jwq.js:177:395)
	(anonymous function) (vendor.min.kek06jwq.js:174)
	(anonymous function) (vendor.min.kek06jwq.js:60)
[Error] ReferenceError: Can't find variable: Notification — capacitor.js:2:27636
	capacitorConsole (user-script:2:78)
	(anonymous function) (user-script:2:356)
	(anonymous function) (user-script:2:376)
[Error] ReferenceError: Can't find variable: Notification
	(anonymous function) (capacitor.js:2:27636)
	(anonymous function) (capacitor.js:2:27191)
	forEach
	(anonymous function) (capacitor.js:2:27111)
	(anonymous function) (capacitor.js:2:27528)
```

**Comment by imhoffd** [comment]
Try using `window.Capacitor.Plugins` for iOS and Android.

This will be much simpler in Capacitor 3 and the documentation will be better.

**Reply by aderowbotham** [reply]
Thank you, I can now call `LocalNotifications.areEnabled()` without getting an error. 

Now the problem is more LocalNotifications-specific because areEnabled() resolves with `value: false`

The plugin doesn't seem to have an `enable()` method. Is this related to whether the capability is added to the app in XCode or do I have to enable it in Javascript too?

**Reply by itshazlan** [reply]
> Try using `window.Capacitor.Plugins` for iOS and Android.
> 
> This will be much simpler in Capacitor 3 and the documentation will be better.

thanks for the solution, works better for calling any method but how about calling enum and options?
